### ISSUE TITLE
Move windows/amd64 to Tier 1 - Primary Support

### ISSUE THREAD

**Issue by pjanotti** [open-telemetry/opentelemetry-collector#11567]
**Is your feature request related to a problem? Please describe.**
`windows/amd64` is the most popular Windows deployment of the collector in production, at this stage we should be able to move it to tier 1 support as defined at https://github.com/open-telemetry/opentelemetry-collector/blob/main/docs/platform-support.md#tier-1--primary-support.

**Describe the solution you'd like**
* CI Test runs on Windows for PRs and release cycle, see related issues:
  - https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/34385
  - https://github.com/open-telemetry/opentelemetry-collector-releases/issues/576

**Comment by djaglowski** [comment]
I am supportive of this in principle but concerned about the limited number of contributors we have supporting windows. My suggestion is that we first enable CI tests for windows on all PRs and use this to build confidence that we're ready to declare this tier 1 support. Maybe we can do this for a month to observe the impact and gauge our ability to respond, and then decide if we're ready to declare tier 1 support?

**Comment by atoulme** [comment]
I agree. I also think we need to enable windows tests for components that are windows specific as they tend to break often.

**Comment by pjanotti** [comment]
On the contrib repo the `scoped-test` action is preventing the merge of broken Windows tests. Many of the flaky tests that fail on Windows are actually concurrency issues that are surfaced by the differences in timing and scheduling on Windows. While the effort to fix the flaky tests will continue I'm going to start also to look at the tests that are being skipped on Windows.

**Comment by douglascamata** [comment]
FYI, we have two issues on the releases repository that are related to this:

- https://github.com/open-telemetry/opentelemetry-collector-releases/issues/1153 (Windows containers are built only on release, not on every commit): I think this one is a **must** for `windows/amd64` to become tier 1. 
- https://github.com/open-telemetry/opentelemetry-collector-releases/issues/1091: This will help us exercise the Windows release pipeline, like we are doing to all the others, to reduce issues during release. I think this one is a **nice to have**.

I think we will solve these fairly soon. Considering there is other work to be done (Windows components tests, build confidence check, and some skipped tests to check), I think (and hope) at least the 1st issue above will be completed by then.

**Comment by atoulme** [comment]
Discussed at SIG meeting on 3/11: there is rough consensus for supporting this change and increase support of Windows to tier 1. Unless there are objections registered by 3/18, a week from now, we will move forward with this change and promote Windows amd64 to tier 1.
